pymaker/dss.py

```diff
@@ -468,48 +468,57 @@ def frob(self, ilk: Ilk, urn_address: Address, dink: Wad, dart: Wad, collateral_
 
     def validate_frob(self, ilk: Ilk, address: Address, dink: Wad, dart: Wad):
         """Helps diagnose `frob` transaction failures by asserting on `require` conditions in the contract"""
+
+        def r(value, decimals=1):  # rounding function
+            return round(float(value), decimals)
+
+        def f(value, decimals=1):  # formatting function
+            return f"{r(value):16,.{decimals}f}"
+
         assert isinstance(ilk, Ilk)
         assert isinstance(address, Address)
         assert isinstance(dink, Wad)
         assert isinstance(dart, Wad)
 
+        assert self.live()  # system is live
+
         urn = self.urn(ilk, address)
         ilk = self.ilk(ilk.name)
+        assert ilk.rate != Ray(0)  # ilk has been initialised
+
+        logger.debug(f"System     | debt {f(self.debt())} | ceiling {f(self.line())}")
+        logger.debug(f"Collateral | debt {f(ilk.art)} | ceiling {f(ilk.line)}")
```

Comment:
System Debt (`self.debt()`) is denominated in DAI, whereas Collateral Debt (`ilk.art`) is denominated in normalized internal DAI. If it would be more helpful to have it denominated in USD (as a proxy for DAI), consider presenting the Collateral's global `tab` instead:
- `tab [USD] = Normalized internal DAI * Collateral Debt Multiplier = ilk.art * ilk.rate`

Reply:
Yes, I should display that to be consistent with the actual check on line 503.  Will commit shortly.